src/factory/ChartFactory.js

```diff
@@ -188,32 +185,41 @@ function ChartFactory({
           description={`${description} ${xDesc}`}
           comparisonData={[]} // TODO: pending NumberVisuals components (HURUmap-UI) fix on this propTypes
           classes={{}} // TODO: pending NumberVisuals style configurations - update root margin
-          {...visualProps}
         />
       );
     }
     case 'grouped_column': {
       const barCount = primaryData[0].length;
       const offset = visualProps.offset || theme.bar.offset;
-      const { domainPadding } = theme.bar;
+      const {
+        domainPadding: {
+          x: [x0, x1]
+        }
+      } = theme.bar;
+      const domainPadding = {
+        x: [x0 * primaryData.length, x1 * primaryData.length]
+      };
       const computedSize =
         primaryData.length * barCount * offset +
         domainPadding.x[0] +
         domainPadding.x[1];
       const height = visualProps.height || theme.bar.height;
-      const computedWidth = horizontal ? height : computedSize;
+      const width = visualProps.width || theme.bar.width;
+      const computedWidth =
+        horizontal || computedSize > width ? width : computedSize;
```

Comment:
We need to do the same check for `computedHeight` as well .... otherwise it doesn't work